src/Backend/SgateShopgatePlugin/Components/User.php

```diff
@@ -80,9 +80,22 @@ public function loginUser($request, $httpResponse)
         $sessionId = $request->getPost('sessionId');
 
         if (isset($sessionId)) {
-            //Set session id using both methods because standard shopware login merges basket with session_id
+            /**
+             * Set session id using both methods because standard shopware
+             * login merges basket with session_id. Setting the session_id
+             * requires resetting the session entirely since it can no longer
+             * be changed once session_start() was called.
+             *
+             * All session variables are saved and replaced once the session
+             * is recreated with the desired id.
+             */
+            $oldSession = array_merge(array(), $_SESSION);
+            session_commit(); // Ends the current session
+            session_id($sessionId); // Sets the ID for the new session
+            session_start(); // Stars a fresh session with the new ID
+            $_SESSION = array_merge(array(), $oldSession); // Replace old session variables
+
```

Comment:
The session ID was being incorrectly set which caused the previous cart items to be lost.